packages/shopware-6-client/src/services/customerService.ts

```diff
@@ -0,0 +1,285 @@
+import { Customer } from "../interfaces/models/checkout/customer/Customer";
+import { apiService } from "../apiService";
+import { CustomerAddress } from "../interfaces/models/checkout/customer/CustomerAddress";
+
+interface CustomerLoginParam {
+  username: string;
+  password: string;
+}
+
+interface CustomerRegisterParam {
+  salutationId: string;
+  firstName: string;
+  lastName: string;
+  password: string;
+  email: string;
+  title?: string;
+  birthdayYear?: number;
+  birthdayMonth?: number;
+  birthdayDay?: number;
+  billingAddress: Partial<
+    BillingAddress
+  > /** TODO: replace Partial with correct optional fields in BillingAddress */;
+  shippingAddress?: ShippingAddress;
+}
+
+interface CustomerUpdateEmailParam {
+  email: string;
+  emailConfirmation: string;
+  password: string;
+}
+
+interface CustomerUpdatePasswordParam {
+  password: string;
+  newPassword: string;
+  newPasswordConfirm: string;
+}
+
+interface CustomerUpdateProfileParam {
+  firstName: string;
+  lastName: string;
+  salutationId: string;
+  title: string;
+}
+
+interface CustomerAddressParam extends Partial<CustomerAddress> {}
+
+interface CustomerLoginResponse {
+  "sw-context-token": string;
+}
+
+interface CustomerRegisterResponse {
+  data: string;
+}
+
+/**
+ * Usage example:
+ * ```ts
+ * import { CustomerService } from '@shopware-pwa/shopware-6-client'
+ * ```
+ */
+export interface CustomerService {
+  login: (params: CustomerLoginParam) => Promise<CustomerLoginResponse>;
+  logout: (contextToken: string) => Promise<null>;
+  register: (
+    params: CustomerRegisterParam
+  ) => Promise<CustomerRegisterResponse>;
+  getCustomer: (contextToken: string) => Promise<Customer> | Promise<never>;
+  getCustomerAddresses: (contextToken?: string) => Promise<CustomerAddress[]>;
+  getCustomerAddress: (
+    addressId: string,
+    contextToken?: string
+  ) => Promise<CustomerAddress>;
+  createCustomerAddress: (
+    params: CustomerAddressParam,
+    contextToken?: string
+  ) => Promise<string>;
+  updateEmail: (params: CustomerUpdateEmailParam) => Promise<null>;
+  updatePassword: (params: CustomerUpdatePasswordParam) => Promise<null>;
+  updateProfile: (params: CustomerUpdateProfileParam) => Promise<null>;
+}
+
+/**
+ * @description Register a customer
+ */
+export const register = async function(
+  params: CustomerRegisterParam
+): Promise<CustomerRegisterResponse> {
+  const resp = await apiService.post(getCustomerEndpoint(), params);
+  return resp.data;
+};
+
+/**
+ * @description Get the context token for current user
+ */
+export const login = async function(
+  params: CustomerLoginParam
+): Promise<CustomerLoginResponse> {
+  const resp = await apiService.post(`${getCustomerEndpoint()}/login`, params);
+  return resp.data;
+};
+
+/**
+ * @description End up the session
+ */
+export const logout = async function(contextToken?: string): Promise<null> {
+  const resp = await apiService.post(`${getCustomerEndpoint()}/logout`, null, {
```

Comment:
we need to have these steps here

1. if contextToken is provided, then call `update({contextToken})`
2. call api here
3. call `update({contextToken: ''})`

don't return anything as it's a void method